src/std/io/exception.d

```diff
@@ -12,9 +12,12 @@ class IOException : Exception
         this.msg = msg.move;
     }
 
-    override void toString(scope void delegate(const scope char[]) @safe sink) const
+    override void toString(scope void delegate(in char[]) sink) const
     {
-        auto nothrowSink = (const scope char[] ch) {
+        // Wrapping `sink` into @trusted is not strictly correct because the sink might escape
+        // the slice (allthough it's highly unlikely)
+        // Retain the old behavior and implement a transition later
+        auto nothrowSink = (const scope char[] ch) @trusted {
```

Comment:
Why is trusted needed here?

Reply:
`ioError` expects a `@safe` delegate.
But I could also remove the `@safe` requirement from that function

Reply:
Oh, I misread! I thought it was calling this right after, but it's later.

Have to look up ioError, and see what it does...

Reply:
Let's remove the `@safe` requirement across the board. I'd rather not leave a bomb like this in there, and you can still use this with a `@safe` delegate. `toString` is already not `@safe` or `@trusted`, so I'm not sure why it requires a `@safe` delegate in the first place.

Reply:
Done.